domain/src/test/java/org/oppia/android/domain/feedbackreporting/FeedbackReportManagementControllerTest.kt

```diff
@@ -0,0 +1,466 @@
+    .setUserSubmittedSuggestion(languageSuggestionText)
+    .build()
+
+  private val userSuppliedSuggestionFeedback = UserSuppliedFeedback.newBuilder()
+    .setSuggestion(featureSuggestion)
+    .build()
+
+  private val laterSuggestionReport = FeedbackReport.newBuilder()
+    .setReportSubmissionTimestampSec(LATER_TIMESTAMP)
+    .setUserSuppliedInfo(userSuppliedSuggestionFeedback)
+    .setAppContext(appContext)
+    .build()
+
+  private val userSuppliedCrashFeedback = UserSuppliedFeedback.newBuilder()
+    .setCrash(Crash.getDefaultInstance())
+    .build()
+
+  private val earlierCrashReport = FeedbackReport.newBuilder()
+    .setReportSubmissionTimestampSec(EARLIER_TIMESTAMP)
+    .setUserSuppliedInfo(userSuppliedCrashFeedback)
+    .setAppContext(appContext)
+    .build()
+
+  @Before
+  fun setUp() {
+    networkConnectionUtil = NetworkConnectionUtil(ApplicationProvider.getApplicationContext())
+    setUpTestApplicationComponent()
+    setUpApplicationForContext()
+    setUpFakeLogcatFile()
+    MockitoAnnotations.initMocks(this)
+  }
+
+  @After
+  fun tearDown() {
+    mockWebServer.shutdown()
+  }
+
+  @Test
+  fun testController_submitFeedbackReport_withNetwork_successfullySendsRequestToServer() {
+    mockWebServer.enqueue(MockResponse().setBody("{}"))
+    networkConnectionUtil.setCurrentConnectionStatus(LOCAL)
+    feedbackReportManagementController.submitFeedbackReport(laterSuggestionReport)
+    // The feedback_reporting_unformatted.json is the same as the feedback_reporting.json, in the
+    // format used in the request body of reports sent in network requests.
+    val feedbackReportUnformattedJson = ApiMockLoader.getFakeJson(
+      "feedback_reporting_unformatted.json"
+    )
+
+    val request = mockWebServer.takeRequest(
+      timeout = testCoroutineDispatcher.DEFAULT_TIMEOUT_SECONDS,
+      unit = testCoroutineDispatcher.DEFAULT_TIMEOUT_UNIT
+    )
+    request?.let {
+      assertThat(it.method).isEqualTo("POST")
+      // Append newline to the request body to match new line at the end of the JSON file.
+      assertThat(it.body.readUtf8() + "\n").isEqualTo(feedbackReportUnformattedJson)
+    }
+  }
+
+  @Test
+  fun testController_submitFeedbackReport_withoutNetwork_doesNotSendRequestToServer() {
+    mockWebServer.enqueue(MockResponse().setBody("{}"))
+    networkConnectionUtil.setCurrentConnectionStatus(NONE)
+    feedbackReportManagementController.submitFeedbackReport(laterSuggestionReport)
+
+    val request = mockWebServer.takeRequest(
+      timeout = testCoroutineDispatcher.DEFAULT_TIMEOUT_SECONDS,
+      unit = testCoroutineDispatcher.DEFAULT_TIMEOUT_UNIT
+    )
+    assertThat(request).isNull()
+  }
+
+  @Test
+  fun testController_submitFeedbackReport_withNetwork_doesNotSaveReportToStore() {
+    // Enqueue a responses so that the MockWebServer knows when the request is successfully sent.
+    mockWebServer.enqueue(MockResponse().setBody("{}"))
+    networkConnectionUtil.setCurrentConnectionStatus(LOCAL)
```

Comment:
Should we add test cases for `CELLULAR` type connections?

Reply:
Done